src/microcosm.js

```diff
@@ -112,7 +112,7 @@ class Microcosm extends Emitter implements Domain {
 
     this.history = this.parent ? this.parent.history : new History(options)
 
-    this.snapshots = {}
+    this.snapshots = Object.create(this.parent ? this.parent.snapshots : null)
```

Comment:
This is new. By using prototypal inheritance, forks gain access to prior snapshots without needing to "sync up" as actions update that they couldn't track. Accessing a snapshot prior to when a fork was created causes prototypal access from the parent.